src/components/ui/EmailForm.tsx

```diff
@@ -28,23 +38,46 @@ export default function ({
             <Email />
           </div>
         }
-        type="email"
-        placeholder={placeholder}
-        value={email}
-        onChange={(e) => setEmail((e.target as HTMLInputElement).value)}
-        onKeyDown={(event) =>
-          event.code === 'Enter' && emailIsValid ? onSubmit(email) : undefined
+        placeholder={emailsAmount ? `${emailsAmount} / 10+` : placeholder}
+        value={inputEmail}
+        valueList={EmailFormStore.emailMapping}
+        removeValueFromList={(fileName, index) =>
+          EmailFormStore.removeEmailsFromList(fileName, index)
+        }
+        onChange={({ target }) =>
+          EmailFormStore.safeInputChecker((target as HTMLInputElement).value)
         }
+        onKeyDown={({ code }) => {
+          if (code === 'Backspace') EmailFormStore.removeLastEmail()
+          if (code === 'Enter' && listIsValid) onSubmit(emailList)
+        }}
       />
+
+      <LoadedFilesList
+        removeValueFromList={(fileName) =>
+          EmailFormStore.removeEmailsFromList(fileName)
+        }
```

Comment:
I guess, you can set it as `removeValueFromList={EmailFormStore.removeEmailsFromList}` without `() =>`

Reply:
Yup, thanks :)